addons/website/static/src/js/menu/content.js

```diff
@@ -1072,8 +1072,90 @@ function _clonePage(pageId) {
     });
 }
 
+const KeyboardNavigationMixin = require('web.KeyboardNavigationMixin');
+const KeyboardNavigationWidget = Widget.extend(KeyboardNavigationMixin, {
+    /**
+     * @override
+     */
+    init() {
+        this.isFullscreen = false;
+        return KeyboardNavigationMixin.init.call(this);
+    },
+    /**
+     * @override
+     */
+    start() {
+        const ret = this._super(...arguments);
+        this.setElement($('body'));
+        return ret;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Toggles the fullscreen mode.
+     *
+     * @private
+     * @param {Boolean} state toggle fullscreen on/off (true/false)
+     */
+    async _toggleFullscreen(state) {
+        this.isFullscreen = state;
+        const dummy = document.createElement('div');
+        const topMenu = document.querySelector('#web_editor-top-edit') || dummy;
+        const leftPanel = document.querySelector('#oe_snippets') || dummy;
+        const navbar = document.querySelector('#oe_main_menu_navbar') || dummy;
+        [topMenu, leftPanel, navbar, document.body].forEach(el => el.style.transition = 'all 0.25s');
+        if (this.isFullscreen) {
+            document.body.style.setProperty('padding', '0', 'important');
+            topMenu.style.setProperty('top', `-${window.getComputedStyle(topMenu).height}`, 'important');
+            navbar.style.setProperty('top', `-${window.getComputedStyle(navbar).height}`, 'important');
+            leftPanel.style.setProperty('left', `-${window.getComputedStyle(leftPanel).width}`, 'important');
+        } else {
+            document.body.style.removeProperty('padding');
+            topMenu.style.removeProperty('top');
+            navbar.style.removeProperty('top');
+            leftPanel.style.removeProperty('left');
+        }
+        let resizing = true;
+        const resizeFunction = () => {
+            if (resizing) {
+                window.dispatchEvent(new Event('resize'));
```

Comment:
Really interesting things here but why dispatch "resize" event on the window ? The window size will not change and even if it did isn't the browser handling that ? Isn't the `requestAnimationFrame` a little overkill ?

Reply:
Because the editor assumes that the #wrapwrap only changes size when the window gets resized, it listens for window resize events to resize covers and other UI elements (including the background position overlay which sbu reported broken). For most applications, the #wrapwrap being resized is effectively the same as the window itself being resized, and I think it's the cleanest way to communicate that to widgets that need to be aware of it, especially since the code is in a completely different subsystem and there is no good way to hook into those behaviours otherwise.

Using requestAnimationFrame allows it to resize during the animation rather than only once when it's over which would jump and defeat the purpose of having an animation in the first place. On that note, I'm just fine with having no animation, especially because there's pretty much 30 lines of JS only for that, but it was a FP/@sbuhl request.